lib/posix/posix_utils.nim

```diff
@@ -99,3 +99,11 @@ proc mkdtemp*(prefix: string): string =
     raise newException(OSError, $strerror(errno))
   return $tmpl
 
+when defined(linux):
+  proc isSsd*(diskLetter: char): bool {.inline.} =
+    ## Returns ``true`` if disk is SSD (Solid). Linux only.
+    ##
+    ## .. code-block:: nim
+    ##   echo isSsd('a') ## 'a' for /dev/sda, 'b' for /dev/sdb, ...
+    ##
+    try: readFile("/sys/block/sd" & $diskLetter & "/queue/rotational") == "0\n" except: false
```

Comment:
`sda`, `sdb` aren't the only things that can be SSD. There are also NVMe drives too.

Also, I don't know how useful would this proc actually be in practice to warrant an addition to the stdlib, given that you have to extract the device letter from whatever an user might give (ie. `/dev/sda`, etc.), and I'm not sure what kind of optimization would target only SSD.

Reply:
But proc is about disk mechanical or solid.

If you find benchmarks of mechanical disk outperforming solid disk link it, I want to read it.
😃

Reply:
I am not convinced that this should be in the stdlib. It appears to be too specialized for a vague use case of "I/O optimization", and it's Linux-only to boot.

I suspect that anyone who want this might already have their own function that do the same, given how trivial it is. IMO this would belong to a 3rd-party package that provides detailed disk information or something, not the stdlib.

Reply:
Yes is Linux-only, but according to the last official Survey, most people are using Linux anyway. 
:man_shrugging: